### PR TITLE
Add copy button, add copiable contract address and query results

### LINKED ISSUE #66: Implement copy button component
A solution for a copy button should be found, to be applied for 

 - contract addresses 
 - code hashes
 - call results

Design missing for addresses and code hashes in Figma, but quite useful. @erler was the omission intentional?

Examples from old Canvas UI: 
Screenshot 2021-08-13 at 18 09 21
Screenshot 2021-08-13 at 18 09 40
Screenshot 2021-08-13 at 18 09 55

### PR DESCRIPTION
**Pull request by kwingram25**
Closes #66 

- Adds copying of addresses when identicon is clicked
- Adds copy button to allow copying to clipboard for contract addresses in header and contract call results
- Refactor TransactionsContext to also handle general UI notifications - currently only clipboard events

### CODE REVIEW

**Comment by statictype** [comment]
please always add a PR description
- what changes in the PR
- what issues does it close

File: src/ui/components/Notifications.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+              text={status}
+            />
+            {isComplete && events && !isEmptyObj(events) && (
+              <Notification
+                dataCy="transaction-complete"
+                icon={<BellIcon className="text-yellow-400 w-12 h-12" />}
+                isDismissable
+                key={`transaction-complete-${id}`}
+                label={Object.keys(events).map(eventName => {
+                  const times = events[eventName] > 1 ? ` (x${events[eventName]})` : '';
```
Comment (statictype) [inline src/ui/components/Notifications.tsx:89]: Using `TransactionContext` which handles on-chain related events and errors for UI notifications seems like the wrong abstraction. Can't we just show a tooltip? There was never a discussion or design in Figma that suggests we should handle them together
please see https://sandimetz.com/blog/2016/1/20/the-wrong-abstraction

File: src/ui/pages/Contract.tsx [hunk 2]
```diff
@@ -50,7 +52,7 @@ export function Contract() {
 
   const [tabIndex, setTabIndex] = useState(TABS.findIndex(({ id }) => id === activeTab) || 1);
 
-  const [isOnChain, setIsOnChain] = useState(true);
+  const [isOnChain, setIsOnChain] = useState<boolean | null>(null);
```
Comment (statictype) [inline src/ui/pages/Contract.tsx:55]: it either is on chain or is not on chain, why does this variable need to be null?

File: src/ui/pages/Contract.tsx [hunk 3]
```diff
@@ -73,22 +75,29 @@ export function Contract() {
   const projectName = contract?.abi.info.contract.name;
 
   return (
-    <Loader isLoading={!contract && isLoading}>
+    <Loader isLoading={(!contract && isLoading) || isNull(isOnChain)}>
```
Comment (statictype) [inline src/ui/pages/Contract.tsx:78]: why should a contract that is not on-chain trigger a loading state for the page? why do we need a loader here anyway? it was probably meant to be for remote fetching but that is not implemented and probably never will be

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -4,14 +4,14 @@ about: Create a report to help us improve
...
 assignees: ''
-
 ---
 
 **Describe the bug**
 A clear and concise description of what the bug is.
 
 **To Reproduce**
 Steps to reproduce the behavior:
+
```
Comment (statictype) [inline .github/ISSUE_TEMPLATE/bug_report.md:14]: unrelated change, please learn to stage your changes properly

File: .github/ISSUE_TEMPLATE/custom.md [hunk 5]
```diff
@@ -4,7 +4,4 @@ about: Describe this issue template's purpose here.
 title: ''
 labels: ''
 assignees: ''
-
 ---
```
Comment (statictype) [inline .github/ISSUE_TEMPLATE/custom.md:7]: same here - unrelated change due to improper change staging

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -24,15 +24,17 @@ A clear and concise description of what you expected to happen.
...
+- OS: [e.g. iOS]
+- Browser [e.g. chrome, safari]
+- Version [e.g. 22]
 
 **Smartphone (please complete the following information):**
- - Device: [e.g. iPhone6]
- - OS: [e.g. iOS8.1]
- - Browser [e.g. stock browser, safari]
- - Version [e.g. 22]
+
```
Comment (statictype) [inline .github/ISSUE_TEMPLATE/bug_report.md:33]: same here - unrelated change due to improper change staging

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 7]
```diff
@@ -4,7 +4,6 @@ about: Suggest an idea for this project
 title: ''
 labels: ''
 assignees: ''
-
```
Comment (statictype) [inline .github/ISSUE_TEMPLATE/feature_request.md:7]: same here - unrelated change due to improper change staging

File: .eslintrc [hunk 8]
```diff
@@ -78,9 +78,9 @@
     "import/newline-after-import": "warn",
     "@typescript-eslint/restrict-template-expressions": "off",
     "react/prop-types": "off",
+    "@typescript-eslint/no-misused-promises": "off",
```
Comment (statictype) [inline .eslintrc:81]: why did you turn off this rule?
this was a question from a previous PR that didn't get an answer

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 9]
```diff
@@ -24,15 +24,17 @@ A clear and concise description of what you expected to happen.
 If applicable, add screenshots to help explain your problem.
 
 **Desktop (please complete the following information):**
- - OS: [e.g. iOS]
- - Browser [e.g. chrome, safari]
- - Version [e.g. 22]
+
```
Comment (statictype) [inline .github/ISSUE_TEMPLATE/bug_report.md:27]: same here - unrelated change due to improper change staging

File: src/ui/components/Notifications.tsx [hunk 10]
```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 @paritytech/contracts-ui authors & contributors
+// SPDX-License-Identifier: GPL-3.0-only
+
+import { BellIcon, XIcon } from '@heroicons/react/outline';
+import { NotificationIcon } from './common/NotificationIcon';
+import type { QueuedTxOptions, NotificationsState, NotificationObject, VoidFn } from 'types';
+import { classes, isEmptyObj, truncate } from 'ui/util';
+
+interface NotificationProps {
+  dataCy?: string;
```
Comment (statictype) [inline src/ui/components/Notifications.tsx:10]: `data-cy` is just another data attribute, part of HTMLAttributes. let's not clutter component props

**Review by statictype — CHANGES_REQUESTED** [review]

File: src/ui/pages/Contract.tsx (hunk 2 above)
Comment (kwingram25) [reply src/ui/pages/Contract.tsx:55]: null indicates the check has not been performed yet

Comment (statictype) [reply src/ui/pages/Contract.tsx:55]: why do we need that? i think `true` is a sensible default. if it turns out that is not on-chain, then we display the "not on-chain" text, otherwise we display a reinstantiate button. keep in mind that the data for the page comes from local storage and URL, regardless of the contract being on chain or not.

File: src/ui/components/Notifications.tsx (hunk 1 above)
Comment (kwingram25) [reply src/ui/components/Notifications.tsx:89]: There were some problems getting a tooltip to show in all cases - mainly when the identicon is clicked, especially in the account dropdown. In general there is still a disconnect between the app design and the need to allow identicon copy on clicking - the contract rows on the homepage all being links but also needing to allow copy on click, etc.

File: src/ui/pages/Contract.tsx (hunk 2 above)
Comment (kwingram25) [reply src/ui/pages/Contract.tsx:55]: I noticed it was causing the page to load as though the contract were still present and then update to show that it wasn't. It's a better user experience in my opinion to just defer showing the final page until it's known one way or another.

File: .eslintrc (hunk 8 above)
Comment (kwingram25) [reply .eslintrc:81]: It started introducing errors unrelated to any changes I had made, even in master. Whatever it was it appears fixed now

File: src/ui/components/Notifications.tsx (hunk 1 above)
Comment (statictype) [reply src/ui/components/Notifications.tsx:99]: not sure I understand what the difficulty is here? can't a simple "copied to clipboard" message show up in another corner of the screen whenever the copy button is clicked? why not ask for clarifications and design direction instead of refactoring the `TransactionsContext` to support clipboard notifications? 
if we are serious about code quality, this change cannot be merged

**Comment by kwingram25** [comment]
Refactored to separate transaction notifications from the "Copied to clipboard" notice - since they have to avoid overlapping I just placed both in an absolutely positioned div over the app body.
